test/cases/parsing/harmony-module-exports/harmony.js

```diff
@@ -0,0 +1,26 @@
+it("has no 'exports' free var", function () {
+	(typeof exports).should.equal("undefined");
+});
+
+it("has no 'module' free var", function () {
+	(typeof module).should.equal("undefined");
+});
+
+it("should throw when accessing 'module.exports'", function () {
+	(function () { module.exports; }).should.throw();
+});
+
+// skipped as they require TDZ support to implement
+// TODO: have an option to enable TDZ (requires 'let' implementation in engine)
+it("throws on access to 'exports'", function () {
+	this.skip(); return;
+	(function () { return exports; }).should.throw(ReferenceError)
+});
+
+it("throws on access to 'module'", function () {
+	this.skip(); return;
+	(function () { module.exports; }).should.throw(ReferenceError)
+});
+
+export function foo () {};
```

Comment:
@sokra just a note: this is marked as an "immutable" export in the generated bundle, but this should be mutable. Is this an optimization if the function isn't modified and/or escapes, or an oversight?

```js
export function foo () { foo = "bar" }
```

The foo's export's `typeof` should change from `'function'` to `'string'` after the first invocation.